File: 2441-largest-positive-integer-that-exists-with-its-negative/Solution.java
// Return the largest positive integer k such that -k also exists in the array

class Solution {
    public int findMaxK(int[] nums) {
        Arrays.sort(nums);

        List<Integer> negatives = new ArrayList<Integer>();

        int i = 0;
        while (i < nums.length) {
            if (nums[i] > 0) { break; }
            negatives.add(nums[i]);
            i++;
        }

        int j = nums.length - 1;
        while (j >= i) {
            if (negatives.contains(nums[j] * -1)) { return nums[j]; }
            j--;
        }

        return -1;
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/Solution.js
// Return the largest positive integer k such that -k also exists in the array

/**
 * @param {number[]} nums
 * @return {number}
 */

var findMaxK = function (nums) {
    nums.sort((a, b) => { return a - b; });

    for (let i = nums.length - 1; i >= 0; i--) {
        if (nums[i] < 0) { break; }
        if (nums.includes(nums[i] * -1)) { return nums[i]; }
    }

    return -1;
};
